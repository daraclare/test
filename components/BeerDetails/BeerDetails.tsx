import React, { FC } from "react";
import styled from "styled-components";
import { BeerType } from "../../types/types";
import { Checkbox } from "../Checkbox/Checkbox";

type BeerDetailsProps = { beer: BeerType };

const StyledDiv = styled.div`
  display: grid;
  grid-template-columns: repeat(auto-fit, minmax(300px, 1fr));
  grid-template-rows: repeat(auto-fit);
`;

const StyledImg = styled.img`
  max-height: 70vh;
  margin: 0 auto;
`;

export const BeerDetails: FC<BeerDetailsProps> = ({ beer }) => {
  const {
    abv,
    description,
    image_url,
    ingredients: { hops, malt, yeast },
    method: { fermentation, mash_temp, twist },
    name,
    tagline
  } = beer;
  const [checkedList, setChecked] = React.useState({});
  const handleChange = (event: any) => {
    setChecked({
      ...checkedList,
      [event.target.id]: !checkedList[event.target.id]
    });
  };
  return (
    <StyledDiv>
      <StyledImg src={image_url} alt={`bottle of ${name}`} />
      <div>
        <h2>
          {name} <span>({tagline})</span>
        </h2>
        <p>ABV: {abv}</p>
        <p>{description}</p>
        <h3>Ingredients</h3>
        <h4>Hops: </h4>
        <ol>
          {hops.map((hopItem: any, index: number) => {
            return (
              <div key={index}>
                <Checkbox
                  name={`${hopItem.name}-${hopItem.add}`}
                  spanText={hopItem.amount.value + hopItem.amount.unit}
                  text={hopItem.name}
                  handleChange={handleChange}
                  isChecked={checkedList[`${hopItem.name}-${hopItem.add}`]}
                />
              </div>
            );
          })}
        </ol>

        <h4>Malt: </h4>
        <ol>
          {malt.map((maltItem: any, index: number) => {
            return (
              <div key={index}>
                <Checkbox
                  name={`${maltItem.name}-${maltItem.add}`}
                  spanText={maltItem.amount.value + maltItem.amount.unit}
                  text={maltItem.name}
                  handleChange={handleChange}
                  isChecked={checkedList[`${maltItem.name}-${maltItem.add}`]}
                />
              </div>
            );
          })}
        </ol>

        <h4>Yeast: </h4>
        <p>{yeast}</p>

        <h3>Methods</h3>
        <p>
          Fermentation: {fermentation.temp.value}
          {fermentation.temp.unit}
        </p>
        <p>Mash: </p>
        {mash_temp.map((mash, index) => {
          return (
            <ul key={index}>
              <li>Duration: {mash.duration} minutes</li>
              <li>
                Temp: {mash.temp.value} {mash.temp.unit}
              </li>
            </ul>
          );
        })}
        <p>Twist?: {twist ? "Yes" : "No"}</p>
      </div>
    </StyledDiv>
  );
};
