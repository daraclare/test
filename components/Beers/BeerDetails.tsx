import React, { FC } from "react";
import styled from "styled-components";
import { BeerType } from "../../types/types";

type BeerDetailsProps = { beer: BeerType };

const StyledDiv = styled.div``;

export const BeerDetails: FC<BeerDetailsProps> = ({ beer }) => {
  console.log("beer", beer);
  const {
    abv,
    description,
    food_pairing,
    id,
    image_url,
    ingredients: { hops, malt, yeast },
    method: { fermentation, mash_temp },
    name,
    tagline
  } = beer;
  return (
    <StyledDiv>
      <h2>
        {name} <span>({tagline})</span>
      </h2>
      <p>{abv}</p>
      <p>{description}</p>
      <h4>Food Pairings: </h4>
      {food_pairing.map((food: string, index: number) => (
        <p key={index}>{food}</p>
      ))}
      <img src={image_url} alt={`bottle of ${name}`} />
    </StyledDiv>
  );
};
