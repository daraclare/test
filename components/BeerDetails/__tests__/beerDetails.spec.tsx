import React from "react";
import { render } from "@testing-library/react";
import { BeerDetails } from "../BeerDetails";
import { BEER_MOCK } from "../__mocks__/beerDetails.mocks";

describe("Component: BeerDetails", () => {
  it("should render without throwing an error", () => {
    const { container } = render(<BeerDetails beer={BEER_MOCK} />);
    expect(container.firstChild).toMatchSnapshot();
  });
  it("should render with correct name, including tagline", () => {
    const { getByTestId } = render(<BeerDetails beer={BEER_MOCK} />);
    const beerName = getByTestId("beer-name");
    expect(beerName.innerHTML).toBe(
      `${BEER_MOCK.name} <span>(${BEER_MOCK.tagline})</span>`
    );
  });
  it("should render with correct abv", () => {
    const { getByTestId } = render(<BeerDetails beer={BEER_MOCK} />);
    const beerAbv = getByTestId("beer-abv");
    expect(beerAbv.innerHTML).toBe(`ABV: ${BEER_MOCK.abv}`);
  });
  it("should render with correct description", () => {
    const { getByTestId } = render(<BeerDetails beer={BEER_MOCK} />);
    const beerDesc = getByTestId("beer-desc");
    expect(beerDesc.innerHTML).toBe(BEER_MOCK.description);
  });
  it("should render with correct number of hop and malt items", () => {
    const { getAllByTestId } = render(<BeerDetails beer={BEER_MOCK} />);
    const ListItem = getAllByTestId("listItem");
    expect(ListItem).toHaveLength(8);
  });
});
