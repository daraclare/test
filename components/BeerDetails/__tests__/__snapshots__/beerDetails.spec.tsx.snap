// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: BeerDetails should render without throwing an error 1`] = `
<div
  class="sc-bxivhb fuKkVT"
>
  <img
    alt="bottle of Buzz"
    class="sc-ifAKCX kbaKWD"
    src="https://images.punkapi.com/v2/keg.png"
  />
  <div>
    <h2
      data-testid="beer-name"
    >
      Buzz
       
      <span>
        (
        A Real Bitter Experience.
        )
      </span>
    </h2>
    <p
      data-testid="beer-abv"
    >
      ABV: 
      4.5
    </p>
    <p
      data-testid="beer-desc"
    >
      A light, crisp and bitter IPA brewed with English and American hops. A small batch brewed only once.
    </p>
    <h3>
      Ingredients
    </h3>
    <h4>
      Hops: 
    </h4>
    <ol>
      <div>
        <li
          class="sc-bdVaJa ibTDPh"
          data-testid="listItem"
        >
          Fuggles
          <span
            class="sc-htpNat kfjJtI"
            data-testid="listItem-spanText"
          >
             
            25grams
             
          </span>
          <button
            class="sc-bwzfXH bGgvvP"
            data-testid="listItem-button"
            id="Fuggles-start"
          >
            IDLE
          </button>
        </li>
      </div>
      <div>
        <li
          class="sc-bdVaJa ibTDPh"
          data-testid="listItem"
        >
          First Gold
          <span
            class="sc-htpNat kfjJtI"
            data-testid="listItem-spanText"
          >
             
            25grams
             
          </span>
          <button
            class="sc-bwzfXH bGgvvP"
            data-testid="listItem-button"
            id="First Gold-start"
          >
            IDLE
          </button>
        </li>
      </div>
      <div>
        <li
          class="sc-bdVaJa ibTDPh"
          data-testid="listItem"
        >
          Fuggles
          <span
            class="sc-htpNat kfjJtI"
            data-testid="listItem-spanText"
          >
             
            37.5grams
             
          </span>
          <button
            class="sc-bwzfXH bGgvvP"
            data-testid="listItem-button"
            id="Fuggles-middle"
          >
            IDLE
          </button>
        </li>
      </div>
      <div>
        <li
          class="sc-bdVaJa ibTDPh"
          data-testid="listItem"
        >
          First Gold
          <span
            class="sc-htpNat kfjJtI"
            data-testid="listItem-spanText"
          >
             
            37.5grams
             
          </span>
          <button
            class="sc-bwzfXH bGgvvP"
            data-testid="listItem-button"
            id="First Gold-middle"
          >
            IDLE
          </button>
        </li>
      </div>
      <div>
        <li
          class="sc-bdVaJa ibTDPh"
          data-testid="listItem"
        >
          Cascade
          <span
            class="sc-htpNat kfjJtI"
            data-testid="listItem-spanText"
          >
             
            37.5grams
             
          </span>
          <button
            class="sc-bwzfXH bGgvvP"
            data-testid="listItem-button"
            id="Cascade-end"
          >
            IDLE
          </button>
        </li>
      </div>
    </ol>
    <h4>
      Malt: 
    </h4>
    <ol>
      <div>
        <li
          class="sc-bdVaJa ibTDPh"
          data-testid="listItem"
        >
          Maris Otter Extra Pale
          <span
            class="sc-htpNat kfjJtI"
            data-testid="listItem-spanText"
          >
             
            3.3kilograms
             
          </span>
          <button
            class="sc-bwzfXH bGgvvP"
            data-testid="listItem-button"
            id="Maris Otter Extra Pale-undefined"
          >
            IDLE
          </button>
        </li>
      </div>
      <div>
        <li
          class="sc-bdVaJa ibTDPh"
          data-testid="listItem"
        >
          Caramalt
          <span
            class="sc-htpNat kfjJtI"
            data-testid="listItem-spanText"
          >
             
            0.2kilograms
             
          </span>
          <button
            class="sc-bwzfXH bGgvvP"
            data-testid="listItem-button"
            id="Caramalt-undefined"
          >
            IDLE
          </button>
        </li>
      </div>
      <div>
        <li
          class="sc-bdVaJa ibTDPh"
          data-testid="listItem"
        >
          Munich
          <span
            class="sc-htpNat kfjJtI"
            data-testid="listItem-spanText"
          >
             
            0.4kilograms
             
          </span>
          <button
            class="sc-bwzfXH bGgvvP"
            data-testid="listItem-button"
            id="Munich-undefined"
          >
            IDLE
          </button>
        </li>
      </div>
    </ol>
    <h4>
      Yeast: 
    </h4>
    <p>
      Wyeast 1056 - American Aleâ„¢
    </p>
    <h3>
      Methods
    </h3>
    <p>
      Fermentation: 
      19
       
      celsius
    </p>
    <div>
      <p>
        Mash Duration: 
        5
         minutes
      </p>
      <p>
        Mash Temp: 
        64
         
        celsius
      </p>
    </div>
    <p>
      Twist?: 
      No
    </p>
  </div>
</div>
`;
